builtin/providers/rundeck/resource_project.go

```diff
@@ -0,0 +1,292 @@
+package rundeck
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/apparentlymart/go-rundeck-api/rundeck"
+)
+
+var projectConfigAttributes = map[string]string{
+	"project.name":                          "name",
+	"project.description":                   "description",
+	"service.FileCopier.default.provider":   "default_node_file_copier_plugin",
+	"service.NodeExecutor.default.provider": "default_node_executor_plugin",
+	"project.ssh-authentication":            "ssh_authentication_type",
+	"project.ssh-key-storage-path":          "ssh_key_storage_path",
+	"project.ssh-keypath":                   "ssh_key_file_path",
+}
+
+func resourceRundeckProject() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateProject,
+		Update: UpdateProject,
+		Delete: DeleteProject,
+		Exists: ProjectExists,
+		Read:   ReadProject,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name for the project",
+				ForceNew:    true,
+			},
```

Comment:
I'm not sure that makes sense here since the name is the primary way users interact with the project in the Rundeck UI, and so an auto-generated project name would make the Rundeck UI unusable as soon as there's more than a couple of them. Does that make sense as a rationale for not doing it? I understand that this way makes it so ``create_before_destroy`` won't work, but I don't think there's any big reason to use that lifecycle model on this resource since Rundeck is generally an operator-facing app rather than an end-user-facing app.